drivers/timer/arcv2_timer0.c

```diff
@@ -205,10 +205,17 @@ void z_clock_set_timeout(s32_t ticks, bool idle)
 
 	/* Round delay up to next tick boundary */
 	delay = ((delay + CYC_PER_TICK - 1) / CYC_PER_TICK) * CYC_PER_TICK;
-	last_load = delay;
 
-	timer0_limit_register_set(last_load - 1);
-	timer0_control_register_set(_ARC_V2_TMR_CTRL_NH | _ARC_V2_TMR_CTRL_IE);
+	if (last_load != delay) {
+		timer0_limit_register_set(delay - 1);
+		if (timer0_control_register_get() && _ARC_V2_TMR_CTRL_IP) {
+			timer0_count_register_set(
+				last_load + timer0_count_register_get());
```

Comment:
As Andy indicated, setting the counter based on reading the counter can lead to losing few ticks and clock drift. Why not follow his approach? Semantics of both timers are similar, I believe.